lib/src/conflicts.rs

```diff
@@ -366,6 +369,45 @@ impl Conflict<Option<FileId>> {
     }
 }
 
+impl<T> Conflict<Option<T>>
+where
+    T: Borrow<TreeValue>,
+{
+    /// If every non-`None` term of a `Conflict<Option<TreeValue>>`
+    /// is a `TreeValue::Tree`, this converts it to
+    /// a `Conflict<Tree>`, with empty trees instead of
+    /// any `None` terms. Otherwise, returns `None`.
+    pub fn to_tree_conflict(
+        &self,
+        store: &Arc<Store>,
+        dir: &RepoPath,
+    ) -> Result<Option<Conflict<Tree>>, BackendError> {
+        let tree_id_conflict = self.maybe_map(|term| match term {
+            None => Some(None),
+            Some(value) => {
+                if let TreeValue::Tree(id) = value.borrow() {
+                    Some(Some(id))
+                } else {
+                    None
+                }
+            }
+        });
+        // TODO: Change to let-else once our MSRV is above 1.65
```

Comment:
You can probably change this to let-else now.

Reply:
I meant to do that, so thanks for the reminder. However, when I tried it just now, I didn't quite like the result (with an early return) any better than the current form, so I just dropped the TODO.